controller/Review.php

```diff
@@ -136,6 +161,110 @@ public function index(): void
         $this->composeView('delegated-view', $data, 'pages/index.tpl', 'tao');
     }
 
+    private function pauseConcurrentSessions(
+        DeliveryExecution $activeExecution
+    ): void {
+        $userIdentifier = $activeExecution->getUserIdentifier();
+
+        if (empty($userIdentifier) || $userIdentifier === LtiUser::ANONYMOUS_USER_URI) {
+            return;
+        }
+
+        $logger = $this->getLogger();
+        $logger->info(
+            sprintf(
+                "%s: Attempting to stop all sessions for user %s",
+                self::class,
+                $userIdentifier
+            )
+        );
+
+        $deliveryExecutionService = $this->getDeliveryExecutionService();
+        $activeExecutionService = $this->getActiveDeliveryExecutionsService();
+
+        $otherExecutionIds = $activeExecutionService->getExecutionIdsForOtherDeliveries(
+            $userIdentifier,
+            $activeExecution->getIdentifier()
+        );
+
+        $count = 0;
+
+        $logger->warning(sprintf("otherExecutionIds: %s", var_export($otherExecutionIds,true)));
+
+        foreach ($otherExecutionIds as $executionId) {
+            try {
+                $execution = $deliveryExecutionService->getDeliveryExecution(
+                    $executionId
+                );
+
+                if (!$execution instanceof DeliveryExecution) {
+                    continue;
+                }
+
+                $logger->debug(
+                    sprintf(
+                        '%s: Current execution %s, pausing non-current execution %s',
+                        self::class,
+                        $activeExecution->getIdentifier(),
+                        $executionId
+                    )
+                );
+
+                $this->pauseSingleExecution($execution);
+                $count++;
+            } catch (Throwable $e) {
+                $logger->warning(
+                    sprintf(
+                        '%s: Unable to pause delivery execution %s: %s',
+                        self::class,
+                        $executionId,
+                        $e->getMessage()
+                    )
+                );
+            }
+        }
+
+        $logger->debug(
```

Comment:
We might be abusing a bit of debug in this very PR, lets check if we can remove some of them. This increases logs capacity and costs in several envs QA AWS, GCP, etc, so it is not activated only on local dev. We have to be mindful on adding logs.